signing/mcl/multisig/vectors_test.go

```diff
@@ -0,0 +1,189 @@
+			returnedVal, err := lls.AggregateSignatures(suite, testVectorCopy.signatures, testVectorCopy.publicKeys)
+			require.Equal(t, testVectorCopy.expectedError, err)
+			require.Equal(t, testVectorCopy.aggregatedSig, returnedVal)
+		})
+	}
+
+}
+
+func TestVerifyAggregatedSigKOSK(t *testing.T) {
+	t.Parallel()
+
+	lls := &BlsMultiSignerKOSK{}
+	err := testData2.GenerateJSONFileKOSKForVerifyAggregatedSigTests(lls)
+	require.Nil(t, err)
+
+	suite := mcl.NewSuiteBLS12()
+	testVectors, err := createTestSetup(suite, "testData/multisigKOSKVerifyAggSig.json")
+	require.Nil(t, err)
+
+	for _, testVector := range testVectors {
+		testVectorCopy := testVector
+
+		t.Run(testVectorCopy.testName, func(t *testing.T) {
+			returnedErr := lls.VerifyAggregatedSig(suite, testVectorCopy.publicKeys, testVectorCopy.aggregatedSig, testVectorCopy.message)
+			require.Equal(t, testVectorCopy.expectedError, returnedErr)
+
+		})
+	}
+
+}
+
+func TestAggregateSignaturesNonKOSK(t *testing.T) {
+	t.Parallel()
+
+	hasher, err := blake2b.NewBlake2bWithSize(blsHashSize)
+	lls := &BlsMultiSigner{
+		Hasher: hasher,
+	}
+	require.Nil(t, err)
+
+	err = testData2.GenerateJSONFileNonKOSKForAggregateSignaturesTests(lls)
+	require.Nil(t, err)
+
+	suite := mcl.NewSuiteBLS12()
+	testVectors, err := createTestSetup(suite, "testData/multisigNonKOSKAggSig.json")
+	require.Nil(t, err)
+
+	for _, testVector := range testVectors {
+		testVectorCopy := testVector
+
+		t.Run(testVectorCopy.testName, func(t *testing.T) {
+			returnedVal, err := lls.AggregateSignatures(suite, testVectorCopy.signatures, testVectorCopy.publicKeys)
+			require.Equal(t, testVectorCopy.expectedError, err)
+			require.Equal(t, testVectorCopy.aggregatedSig, returnedVal)
+		})
+	}
+
+}
+
+func TestVerifyAggregatedSigNonKOSK(t *testing.T) {
+	t.Parallel()
+
+	hasher, err := blake2b.NewBlake2bWithSize(blsHashSize)
+	require.Nil(t, err)
+	lls := &BlsMultiSigner{
+		Hasher: hasher,
+	}
+
+	err = testData2.GenerateJSONFileNonKOSKForVerifyAggregatedSigTests(lls)
+	require.Nil(t, err)
+
+	suite := mcl.NewSuiteBLS12()
+	testVectors, err := createTestSetup(suite, "testData/multisigNonKOSKVerifyAggSig.json")
+	require.Nil(t, err)
+
+	for _, testVector := range testVectors {
+		testVectorCopy := testVector
+
+		t.Run(testVectorCopy.testName, func(t *testing.T) {
+			returnedErr := lls.VerifyAggregatedSig(suite, testVectorCopy.publicKeys, testVectorCopy.aggregatedSig, testVectorCopy.message)
+			require.Equal(t, testVectorCopy.expectedError, returnedErr)
+		})
+	}
+
+}
+
+func createTestSetup(suite crypto.Suite, filename string) ([]TestVector, error) {
+	var testVectors []TestVector
+	kg := signing.NewKeyGenerator(suite)
+
+	content, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	var jsonContent testData2.JSONFileContent
+	err = json.Unmarshal(content, &jsonContent)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, testVector := range jsonContent.TestVectors {
+
+		testName := testVector.TestName
```

Comment:
can you extract the lines 149 -> 175 into a function that creates a TestVector instance?

Reply:
updated